tests/test_news_flash.py

```diff
@@ -268,3 +320,40 @@ def test_classification_statistics_ynet():
     assert precision > BEST_PRECISION_YNET
     assert recall > BEST_RECALL_YNET
     assert f1 > BEST_F1_YNET
+
+
```

Comment:
There's a better and more idiomatic way of coding this pattern. Use [contextlib.contextmanager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager). Roughly:

```python
@contextmanager
def insert_waze_alert_to_db():
    try:
        ... # create accident alert
        yield waze_alert
    finally:
        ... # delete waze alert
```

Client code will use a `with` statement.

Actually, and probably even better, would be using pytest's [fixtures](https://docs.pytest.org/en/stable/fixture.html#fixtures-as-function-arguments) for that. But no other test use them, so I'd put that in a TODO.

Reply:
Looking at this, I have an unrelated Python question: Assuming we are not using threads, why use `yield`?

Reply:
Yield in python has nothing to do with threading. It's about separating different control flow logics. It is a concise way of writing iterator objects (generators) using function-like syntax. In this example it simply splits the function to two phases, "enter" and "exit".